src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,162 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static final String PASSWORD_CORRECT = "Abcdfg";
+    private static final String PASSWORD_INCORRECT = "dfg";
+    private static final int AGE_CORRECT = 19;
+    private static final int AGE_BOUND = 18;
+    private static final int AGE_INCORRECT = 17;
+    private static final int AGE_NEGATIVE = -18;
+    private static RegistrationService service;
+    private static StorageDao storageDao;
+    private User actual;
+
+    @BeforeAll
+    static void beforeAll() {
+        service = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @BeforeEach
+    void setUp() {
+        actual = new User();
+    }
+
+    @Test
+    void register_nullUser_throwsException() {
+        Exception exception = assertThrows(RuntimeException.class,
+                () -> service.register(null), "An Exception must be thrown here");
+        String expectedExceptionMessage = "User is null.";
+        assertEquals(expectedExceptionMessage,
+                exception.getMessage(),
+                "Different exception messages");
+    }
+
+    @Test
+    void register_nullLogin_throwsException() {
+        Exception exception = assertThrows(RuntimeException.class,
+                () -> service.register(actual), "An Exception must be thrown here");
+        String expectedExceptionMessage = "User login is null.";
+        assertEquals(expectedExceptionMessage,
+                exception.getMessage(),
+                "Different exception messages");
+    }
+
+    @Test
+    void register_newUser_ok() {
+        actual.setLogin("user0");
+        actual.setPassword(PASSWORD_CORRECT);
+        actual.setAge(AGE_CORRECT);
+        service.register(actual);
+        assertEquals(storageDao.get(actual.getLogin()), actual, "User hasn't been added");
+    }
+
+    @Test
+    void register_existingUser_throwsException() {
+        actual.setLogin("user1");
+        actual.setPassword(PASSWORD_CORRECT);
+        actual.setAge(AGE_CORRECT);
+        service.register(actual);
+        Exception exception = assertThrows(RuntimeException.class,
+                () -> service.register(actual), "An Exception must be thrown here");
+        String expectedExceptionMessage = "User with login "
+                + actual.getLogin() + " is already exists in the database.";
+        assertEquals(expectedExceptionMessage,
+                exception.getMessage(),
+                "Different exception messages");
```

Comment:
```suggestion
        assertThrows(RuntimeException.class,
                () -> service.register(actual), "An Exception must be thrown here");
```
This will be enough. Fix everywhere

Reply:
Done